tests/test_api_v1_system_crl.py

```diff
@@ -0,0 +1,128 @@
+import e2e_test_framework
+import base64
+
+
+class APIE2ETestSystemCRL(e2e_test_framework.APIE2ETest):
+    uri = "/api/v1/system/crl"
+    crl_text = "-----BEGIN X509 CRL----- \
+MIICnzCCAYcCAQEwDQYJKoZIhvcNAQEFBQAwgYUxKjAoBgNVBAMTIWludGVybmFs \
+LWNhLXVuaXQtdGVzdC5leGFtcGxlLmNvbTELMAkGA1UEBhMCVVMxDTALBgNVBAgT \
+BFV0YWgxFzAVBgNVBAcTDlNhbHQgTGFrZSBDaXR5MRUwEwYDVQQKEwxUZXN0IENv \
+bXBhbnkxCzAJBgNVBAsTAklUFw0yMjAxMTYxMDU3MTJaFw00OTA2MDIxMDU3MTJa \
+oIHMMIHJMIG5BgNVHSMEgbEwga6AFO1y5+VKpptoGIfaSmV4qC+4OmjVoYGLpIGI \
+MIGFMSowKAYDVQQDEyFpbnRlcm5hbC1jYS11bml0LXRlc3QuZXhhbXBsZS5jb20x \
+CzAJBgNVBAYTAlVTMQ0wCwYDVQQIEwRVdGFoMRcwFQYDVQQHEw5TYWx0IExha2Ug \
+Q2l0eTEVMBMGA1UEChMMVGVzdCBDb21wYW55MQswCQYDVQQLEwJJVIIIBnad7uJF \
+O0owCwYDVR0UBAQCAicQMA0GCSqGSIb3DQEBBQUAA4IBAQAAQ7RmrztHk+CjOlAZ \
+eJ8CB5kpzFA5GaGXNGXipzZbwkeWUHmvx6Q5PoFOtCzX+2xaYnQUAQE5IAnBPKMj \
+QeU+nzUk25jrsrVxgow9n/ZLKL5178/5HP6h0Bks90x7IwgaZiWMK4wip0t9f2N4 \
+DGCqxKKphijOPLYknZOtGOUhir3ZGQit5UIF1XJwGKyw0evc4u3lwIGtHMCFa9Zo \
+YVmJzk1NE+UiENtk1GXZO9R/0/PdyskeeZA4gC0XDNWXqD/kvVc5kFHicX1SRk27 \
+BEswL+tABUNMaIVoGkVPSzlnSzHqEIVwC23S4w34o2pQUP0DRdhFaA+v21cAsBNa \
+4bV6 \
+-----END X509 CRL----- \
+"
+    get_tests = [{"name": "Read system CRLs"}]
+    post_tests = [
+        {
+            "name": "Create internal CRL",
+            "payload": {
+                "caref": "61c410f04b782",
```

Comment:
E2E tests can and should create dependencies from other endpoints. Let's make sure the test also creates the required CA before creating the CRL and then ensure that the CA is also removed after all CRL tests are done. Otherwise, E2E tests for this endpoint will fail unless the dependency is manually created. 

A good example of an E2E test creating dependencies can be seen [here](https://github.com/jaredhendrickson13/pfsense-api/blob/v140/tests/test_api_v1_system_certificate.py). Note how the internal CA is created before testing creation of internal certificates using the CA.

Reply:
Added creation of CA certificate dependency in E2E test.
The example was very useful thanks.